### PR TITLE
Add file path join to stdlib

### PR DESCRIPTION
**Pull request by wildum**
#### PR Description

Adding this function to the stdlib is needed to move forward with the [relative path solution](https://github.com/grafana/alloy/pull/789) that is implemented for the modules. This function should also be useful in other scenarios.

I put this function in a new namespace called "file". LMK if you would prefer this function to be in another namespace or without a namespace.

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Review by tpaschalis — APPROVED** [review]

File: syntax/vm/vm_stdlib_test.go [hunk 1]
```diff
@@ -215,6 +215,30 @@ func TestStdlib_StringFunc(t *testing.T) {
...
 }
 
+func TestStdlibFileFunc(t *testing.T) {
+	tt := []struct {
+		name   string
+		input  string
+		expect interface{}
+	}{
+		{"file.path_join", `file.path_join("this/is", "a/path")`, "this/is/a/path"},
+		{"file.path_join empty", `file.path_join()`, ""},
```
Comment (thampiotr) [inline syntax/vm/vm_stdlib_test.go:225]: Is this going to work on Windows where paths use \ ?

Comment (wildum) [reply syntax/vm/vm_stdlib_test.go:225]: yeah it should, it uses the filepath.Join function which is using an OS specific separator (https://pkg.go.dev/path/filepath#Join)

File: docs/sources/reference/stdlib/file.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+title: file
+---
+
+# file
+
+The `file` namespace contains functions related to files.
+
+## file.path_join
+
+The `file.path_join` function joins any number of path elements into a single path, separating them with an OS specific separator.
```
Comment (clayton-cornell) [inline docs/sources/reference/stdlib/file.md:14]:
```suggestion
The `file.path_join` function joins any number of path elements into a single path, separating them with an OS-specific separator.
```

File: docs/sources/reference/stdlib/file.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+### Examples
+
+```
+> file.path_join()
+""
+
+> file.path_join("this/is", "a/path")
```
Comment (clayton-cornell) [inline docs/sources/reference/stdlib/file.md:22]:
```suggestion
file.path_join()
""

file.path_join("this/is", "a/path")
```

Is the greater-than symbol representing the OK command line or part of the actual command.

Comment (wildum) [reply docs/sources/reference/stdlib/file.md:22]: we use it in the other stdlib doc, I guess it's just a style to show input and output

**Pull request merged**